kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/publisher/CloudPublisher.java

```diff
@@ -280,9 +285,17 @@ private void publish(final List<WireRecord> wireRecords) {
 
     private Map<String, Object> buildKuraMessageProperties(final WireRecord wireRecord) {
         Map<String, TypedValue<?>> wireRecordProps = wireRecord.getProperties();
+        Set<String> metricProperties = new HashSet<>(wireRecordProps.keySet());
```

Comment:
Can you refactor this method? It is quite cryptic. Maybe add more meaningful names to variables.

Reply:
Updated with the same logic of metrics: first adding all the metrics to the to the Map, then remove only if the flag is true.